### PR TITLE
docs(rlog): Add a chapter about RLOG

### PR DESCRIPTION
**Pull request by k32**

### CODE REVIEW

File: en_US/advanced/cluster.md [hunk 1]
```diff
@@ -341,3 +341,53 @@ also be allowed by the firewall. The port mapping rule is similar to the node di
...
+While being blazing-fast in the small clusters, it has some scalability issues when the number of nodes in the cluster becomes large.
+Also it becomes more vulnerable to network splits.
+To address these issues, EMQ X supports an experimental extension to Mnesia called `RLOG`.
+This extension can be used in large EMQ X clusters (4 nodes and more).
+
+### RLOG principles
+
+RLOG extension works by separating nodes in the cluster into two logical groups:
+
+- core nodes
```
Comment (qzhuyan) [inline en_US/advanced/cluster.md:363]: describe the number of core nodes, required, suggested?

having only one core node can work?

File: en_US/advanced/cluster.md [hunk 2]
```diff
@@ -341,3 +341,53 @@ also be allowed by the firewall. The port mapping rule is similar to the node di
...
+
+![image](./assets/rlog_cluster.png)
+
+### Enabling RLOG feature
+
+Let's suppose we delegate `core` role to nodes `emqx@core1.internal` and `emqx@core2.internal`.
+
+The following changes should be done in the `etc/emqx.conf` file:
+
+1. Set `cluster.db_backend` parameter to `rlog` on all nodes in the cluster
```
Comment (qzhuyan) [inline en_US/advanced/cluster.md:398]: is it for provsioing new cluster?

clarivify if migration is supported or not.

File: en_US/advanced/cluster.md [hunk 3]
```diff
@@ -341,3 +341,53 @@ also be allowed by the firewall. The port mapping rule is similar to the node di
...
+It is possible to put replicate nodes into an auto-scaling group.
+
+::: warning
+It takes time for the new replicant to join the cluster, and this process generates some additional load on the core nodes.
+Don't use too aggressive autoscaling settings to avoid frequent replicant restarts.
+:::
+
+Flow of the data in the cluster is illustrated in the following diagram:
+
+![image](./assets/rlog_cluster.png)
```
Comment (qzhuyan) [inline en_US/advanced/cluster.md:389]: maybe describe how replicant node select the core node to transfer the data?

File: en_US/advanced/cluster.md (hunk 1 above)
Comment (k32) [reply en_US/advanced/cluster.md:363]: Done.

File: en_US/advanced/cluster.md (hunk 3 above)
Comment (k32) [reply en_US/advanced/cluster.md:389]: Thanks for the remark, added.

File: en_US/advanced/cluster.md (hunk 2 above)
Comment (k32) [reply en_US/advanced/cluster.md:398]: Added a bullet for stopping the cluster.

**Review by qzhuyan — APPROVED** [review]

**Pull request merged**